glide-core/src/client/reconnecting_connection.rs

```diff
@@ -119,7 +120,14 @@ async fn create_connection(
     discover_az: bool,
     connection_timeout: Duration,
 ) -> Result<ReconnectingConnection, (ReconnectingConnection, RedisError)> {
-    let client = &connection_backend.connection_info;
+    let client = {
+        let guard = connection_backend
+            .connection_info
+            .read()
+            .expect("READ_LOCK_ERR");
```

Comment:
you should define consts and use them:
const WRITE_LOCK_ERR: &str = "Failed to acquire the write lock";
const READ_LOCK_ERR: &str = "Failed to acquire the read lock";